x/gamm/pool-models/stableswap/amm.go

```diff
@@ -441,9 +441,24 @@ func (p *Pool) calcSingleAssetJoinShares(tokenIn sdk.Coin, swapFee sdk.Dec) (sdk
 		return &paCopy
 	}
 
-	// We apply the swap fee by multiplying by (1 - swapFee) and then truncating to int
-	oneMinusSwapFee := sdk.OneDec().Sub(swapFee)
-	tokenInAmtAfterFee := tokenIn.Amount.ToDec().Mul(oneMinusSwapFee).TruncateInt()
+	// Find the portion of the pool made up by non-input assets
+	sumScalingFactors := int64(0)
```

Comment:
Not sure if there are more reasons for performing these conversions but if there aren't any, we could just use `uint64` throughout function

```suggestion
	sumScalingFactors := uint64(0)
```